PyDSS/pyPostprocessor/PostprocessScripts/AutomatedThermalUpgrade.py

```diff
@@ -186,21 +183,61 @@ def __init__(self, project, scenario, inputs, dssInstance, dssSolver, dssObjects
 
         # Mitigate thermal violations
         self.Line_trial_counter = 0
-        while len(self.xfmr_violations) > 0 or len(self.line_violations) > 0:
+        # TODO change to 20
+        self.max_upgrade_iteration = min(5, len(self.xfmr_violations) + len(self.line_violations))
+        while (len(self.xfmr_violations) > 0 or len(self.line_violations) > 0) \
+                and (self.Line_trial_counter < self.max_upgrade_iteration):
+            prev_xfmr = len(self.xfmr_violations)
+            prev_line = len(self.line_violations)
             self.dssSolver.Solve()
+            print(f"Solution Converged: {dss.Solution.Converged()}")

```

Comment:
Should these print statements be converted to log statements? In print statements are recording the same information as log statements and so they can be removed. Console and/or file logging can be enabled at different levels of verbosity.

Reply:
Print statements removed and converted to log statements if not duplicated.